### PR TITLE
daemon: fix overwritten files

### PR DESCRIPTION
**Pull request by rphillips**
Files within ignition can be overwritten by later entries. This fixes
the validation to check all ignition written files for validity instead
of checking only the first.

This fixes an issue where the MCD marks a node as degraded when a file within ignition has overwritten a default config (ie: KubeletConfig)

/cc @jeremyeder @sjenning

### CODE REVIEW

File: pkg/daemon/daemon.go [hunk 1]
```diff
@@ -847,7 +847,8 @@ func (dn *Daemon) checkUnits(units []ignv2_2types.Unit) bool {
 
 // checkFiles validates the contents of  all the files in the
 // target config.
-func (dn *Daemon) checkFiles(files []ignv2_2types.File) bool {
+func checkFiles(files []ignv2_2types.File) bool {
+	overwrittenFiles := make(map[string]bool)
```
Comment (jlebon) [inline pkg/daemon/daemon.go]: Though `checkFileContentsAndMode` will still be spouting scary messages when it handles the initial instance of a path. WDYT about just reversing the list/iterating backwards and skipping over paths we've already checked?

File: pkg/daemon/daemon.go [hunk 2]
```diff
@@ -22,6 +22,7 @@ import (
 	mcfgclientv1 "github.com/openshift/machine-config-operator/pkg/generated/clientset/versioned/typed/machineconfiguration.openshift.io/v1"
 	"github.com/pkg/errors"
 	"github.com/vincent-petithory/dataurl"
+	fsnotify "gopkg.in/fsnotify.v1"
```
Comment (jlebon) [inline pkg/daemon/daemon.go]: Minor/optional: this is from `gofmt` I suppose? We usually try to split those out as separate commits to make review easier.

File: pkg/daemon/daemon.go (hunk 1 above)
Comment (rphillips) [reply pkg/daemon/daemon.go]: will do.

**Comment by rphillips** [comment]
@jlebon Pushed the requested fixes... Thank you for the review!
